### DISCUSSION TITLE
Remove puzzles with unclear solutions

### DISCUSSION THREAD

**Discussion by danbockapps** [chesspecker/chesspecker#2] [General]
I absolutely love the site. One request: please filter out the bad Lichess puzzles that have really unclear solutions. For example, the "solution" to this puzzle is a position where black is still down a piece: https://lichess.org/training/MPkjk

It should be quite straightforward with chess.js or similar library to check the solutions and make sure it's either mate or our side has won some material. I'm willing to help with this if I can get some pointers.

**Comment by pnodet** [comment]
Hi! Thanks for the support.

Puzzle should have clear solutions.

Using the Lichess puzzle database, we ensure that the puzzles generated meet a certain standard. When they are created, they are compared to an exhaustive list of criteria. However, being processed by an engine, it's not always clear where the advantage is. In this case stockfish gives roughly a 0.0 eval on move 14 and -4.8 after the puzzle.

My guess is that stockfish "thinks" that black has won a decisive advantage and down a piece but up three pawn black seems to have a lot of compensation!

We do use chess.js to check that a mate on the board is always a win though.

**Reply by danbockapps** [reply]
I know there are a lot of criteria in creating a puzzle. I'm saying one rule should be added: the solution must involve mate or winning material.

In that position with black having three pawns for a piece, you have to be a master or a computer to know for sure that black is winning. That's not a good solution for a puzzle. Don't you agree? Shouldn't puzzles be accessible to non-masters? Shouldn't they have clear solutions that don't require computer analysis to understand?

**Reply by PairOfTrousers** [reply]
Positional puzzles are a very valuable learning resource. If you'd prefer not to see them, I'd recommend using the motifs feature to select things like checkmates only.

**Comment by cshancock** [comment]
I like having these sorts of puzzles; resilient defending is a worthwhile skill!

One way to improve the puzzles available in Chesspecker (which may have already been done) is to filter out those puzzles that have a Popularity rating below some threshold - say, 1 standard deviation below the mean of the selected puzzle theme. You could also require a certain value in the NbPlays field in the puzzle database. 

Of course it could be the case that Lichess users dislike these sorts of puzzles as well, so they score poorly in the Popularity field!

**Reply by pnodet** [reply]
This is not implemented yet but could be done indeed, needs some thinking & testing!
